src/nginx/module.cc

```diff
@@ -49,6 +49,41 @@ namespace google {
 namespace api_manager {
 namespace nginx {
 
+namespace {
+
+// Updates r->uri and r->unparsed_uri, based on api basepath math.
+// Then returns ApiManager::ApiBasepathRewriteAction.
+// If the return value is ApiManager::ApiBasepathRewriteAction::REJECT,
+// then nginx should response status 404 NotFound
+ApiManager::ApiBasepathRewriteAction ngx_esp_rewrite_uri(
+    ngx_http_request_t *r, ngx_esp_loc_conf_t *lc) {
+  std::string unparsed_uri;
+  ApiManager::ApiBasepathRewriteAction unparsed_uri_action =
+      lc->esp->ReWriteURL(ngx_str_to_std(r->unparsed_uri), &unparsed_uri);
+
+  if (unparsed_uri_action == ApiManager::ApiBasepathRewriteAction::REWRITE) {
+    ngx_str_copy_from_std(r->pool, unparsed_uri, &r->unparsed_uri);
+    ngx_str_copy_from_std(r->pool,
+                          ngx_str_to_std(r->method_name) + " " + unparsed_uri +
+                              " " + ngx_str_to_std(r->http_protocol),
+                          &r->request_line);
+  } else if (unparsed_uri_action ==
+             ApiManager::ApiBasepathRewriteAction::REJECT) {
```

Comment:
should we just "return" for  None case?

Reply:
I think if action == NONE,  we should return it too.  Only REWRITE case, we need to run the rest of code.

So the code at line 64 can be:

if (uri_action != REWRITE) return uri_action;

Reply:
My bad. You are right. I updated.